istioctl/pkg/rbac/upgrade_test.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rbac
+
+import (
+	"reflect"
+	"testing"
+)
+
+const (
+	testFailedWithError   = "test failed with error %v"
+	testFailedExpectedGot = "test failed, expected %s, got %s"
+)
+
+type testCases struct {
+	input    Upgrader
+	expected string
+}
+
+func TestUpgradeLocalFile(t *testing.T) {
+	upgrader := Upgrader{
+		RbacFile:             "./testdata/rbac-policies.yaml",
+		RoleToWorkloadLabels: map[string]ServiceToWorkloadLabels{},
+	}
+	// Data from the BookExample. productpage.svc.cluster.local is the service with pod label
+	// app: productpage.
+	upgrader.RoleToWorkloadLabels["service-viewer"] = ServiceToWorkloadLabels{
```

Comment:
This should be part of the test case, and we should also test if the service role doesn't match to any of the `ServiceToWorkloadLabels`. This might be an error from the user and we might just stop and show an error message on this.

Reply:
It is part of the test case. https://github.com/istio/istio/pull/13627/files/922574d4eba75389f6393e65901ba729c7e85872#diff-99e7ee7f0650e963b304225be2ce402aR46

I will add some more tests and do a little refactoring.

Reply:
I mean it should be defined inside the test case so that we can easily test if the mapping doesn't exist, you can do this in follow up PRs

Reply:
Or do you mean this upgrader mock should be part of every test case?

Reply:
I think the second case (part of every test case) makes more sense?

Reply:
Oh ok GitHub didn't refresh fast enough. Cool.